src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -510,6 +515,11 @@ static function (): bool {
                     }
                 );
             }
+
+            $this->lastActivityTime = time();
+        } elseif (isset($this->connectionOptions['heartbeat']) && time() > $this->lastActivityTime + ($this->connectionOptions['heartbeat'] * 2)) {
```

Comment:
If hearbeat options is set to 0 it means that the connection will be unlimited. But with this code, the connection will be reconnect everytime. You can change your condition to ensure that a heartbeat set to 0 do not reset the connection. Thanks !